lightning/src/ln/peer_handler.rs

```diff
@@ -64,6 +64,20 @@ pub trait CustomMessageHandler: wire::CustomMessageReader {
 	/// in the process. Each message is paired with the node id of the intended recipient. If no
 	/// connection to the node exists, then the message is simply not sent.
 	fn get_and_clear_pending_msg(&self) -> Vec<(PublicKey, Self::CustomMessage)>;
+
+	/// Gets the node feature flags which this handler itself supports. All available handlers are
+	/// queried similarly and their feature flags are OR'd together to form the [`NodeFeatures`]
+	/// which are broadcasted in our [`NodeAnnouncement`] message.
+	///
+	/// [`NodeAnnouncement`]: crate::ln::msgs::NodeAnnouncement
+	fn provided_node_features(&self) -> NodeFeatures;
+
+	/// Gets the init feature flags which should be sent to the given peer. All available handlers
+	/// are queried similarly and their feature flags are OR'd together to form the [`InitFeatures`]
+	/// which are sent in our [`Init`] message.
+	///
+	/// [`Init`]: crate::ln::msgs::Init
+	fn provided_init_features(&self, their_node_id: &PublicKey) -> InitFeatures;
```

Comment:
Mentioned it briefly in review club - I'm wondering why there's only `provided_x_features` for `node` and `init` in the various message handlers and no `provided_channel_features`? I see in the `channelmanager` module there's `provided_channel_features` but its just wrappers around `provided_init_features` with the user config. Is this just because right now there aren't much use for the channel announcement message to communicate features?

Reply:
Possibly but will let @TheBlueMatt confirm.

Reply:
Ah gotcha, thanks